rails/app/controllers/portal/offerings_controller.rb

```diff
@@ -163,7 +163,9 @@ def report
     authorize offering
     report = DefaultReportService::default_report_for_offering(offering)
     raise ActionController::RoutingError.new('Default Report Not Found') unless report
-    next_url = report.url_for_offering(offering, current_visitor, request.protocol, request.host_with_port, { activity_id: activity_id })
+    additional_params = { activity_id: activity_id }
+    additional_params[:researcher] = 'true' unless current_user.has_full_access_to_student_data?(offering.clazz)
```

Comment:
Same comment here - can't we just add the `researcher=true` param when we iterate over the external report urls to create the buttons in the researcher class view?